attestation-agent/attestation-agent/src/lib.rs

```diff
@@ -70,21 +75,50 @@ pub trait AttestationAPIs {
 /// Attestation agent to provide attestation service.
 pub struct AttestationAgent {
     _config: Config,
+    eventlog: EventLog,
 }
 
 impl Default for AttestationAgent {
     fn default() -> Self {
         if let Ok(_config) = Config::try_from(config::DEFAULT_AA_CONFIG_PATH) {
-            return AttestationAgent { _config };
+            let eventlog =
+                EventLog::new(&_config.eventlog_config.eventlog_path).expect("eventlog initialize");
+            return AttestationAgent { _config, eventlog };
         }
 
+        let eventlog = EventLog::new(DEFAULT_EVENTLOG_PATH).expect("eventlog initialize");
+
         AttestationAgent {
             _config: Config::default(),
+            eventlog,
         }
     }
 }
 
 impl AttestationAgent {
+    pub async fn init(&self) -> Result<()> {
+        let init_entry = match self._config.eventlog_config.eventlog_algorithm {
+            HashAlgorithm::Sha256 => "INIT sha256/0000000000000000000000000000000000000000000000000000000000000000",
+            HashAlgorithm::Sha384 => "INIT sha384/000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+            HashAlgorithm::Sha512 => "INIT sha512/00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
+        };
+
+        let event_digest = self
+            ._config
+            .eventlog_config
+            .eventlog_algorithm
+            .digest(init_entry.as_bytes());
+        let tee_type = detect_tee_type();
+        let attester = TryInto::<BoxedAttester>::try_into(tee_type)?;
+
+        attester
+            .extend_runtime_measurement(event_digest, DEFAULT_PCR_INDEX)
+            .await
+            .context("write INIT entry")?;
+
```

Comment:
we need `write_log` for `init_entry` here to allow remote verifier to get the digest algorithm and reproduce the rtmr digest

Reply:
Yes, thanks!